### DISCUSSION TITLE
DRY arguments without creating a dedicated input

### DISCUSSION THREAD

**Discussion by ghost** [strawberry-graphql/strawberry#1095] [Q&A] [answered]
Hey, when creating a (relay) connection field I currently need to specify the "after", "before", "first" and "last" argument over and over for each connection.

Is there a way to do this in a dry way (without abstracting those arguments into a dedicated input)?

Example:
```
@strawberry.type
class Query:
@strawberry.field
    def products(
        self,
        after: Optional[ID] = None,
        before: Optional[ID] = None,
        first: Optional[int] = None,
        last: Optional[int] = None,
    ) -> Connection[Product]:
    …
```

I want to be able to still query the products like this:
```
query products(first: 10, …)…
```

**Comment by patrick91** [comment, marked as answer]
@nilsdet I think this is something that's covered by @jkimbo's proposal: https://github.com/strawberry-graphql/strawberry/issues/920

Not supported yet, but should be soon 😊

**Reply by ghost** [reply]
Awesome! Thanks
